content/docs/concepts/architecture/index.md

```diff
@@ -9,73 +9,137 @@ Keptn is an event-based control plane for continuous delivery and automated oper
-
-- **remediation-service:** It is used to determine the action to be performed in remediation workflows. 
-
-Execution plane services subscribe to events via a distributor sidecar, that forwards incoming `.triggered` events to them. These distributor sidecars can also be used to send `.started` and `.finished` events back to the Keptn control plane.
-
-As illustrated in the architecture diagram above, execution plane services can both be operated within the same cluster as Keptn, or outside of the cluster. 
-
- - When they are operated within the same cluster, the services can directly access the HTTP APIs provided by the control plane services,
-without having to authenticate. In this case, the distributor sidecars directly connect themselves to the NATS cluster to subscribe to topics and send back events.
-
-- When an execution plane is operated outside of the Cluster, it has the possibility to communicate with the HTTP API exposed by the `api-gateway-nginx`. In this case, every request to the API has to be authenticated using the `keptn-api` token. 
-Also, the distributor sidecars do not have the possibility to directly connect to the NATS cluster, but they can be configured to fetch open `.triggered` events from the HTTP API.
-
-To read more about developing execution plane services, please refer to the following section in the docs: [Write a Keptn-service](../../0.15.x/integrations/custom_integration/), which helps you to implement a custom service for Keptn. 
+The Keptn Execution Plane hosts the Keptn-services
+that integrate the tools that are used to process the tasks.
+The Keptn cluster can contain a single Execution Plane
+that is installed either in the same Kubernetes cluster as the Control Plane
+or in a different Kubernetes cluster..
+You can also configure multiple Execution Planes on multiple Kubernetes clusters.
+
+Keptn-services react to `.triggered` Keptn events that are sent by the shipyard controller.
+They perform continuous delivery tasks like deploying or promoting a service
+and orchestrational tasks for automating operations.
+Those services can be easily plugged into a task sequence
+to extend the delivery pipeline or to further automate operations.
+Execution plane services subscribe to events via a
+[distributor](../../0.18.x/reference/miscellaneous/distributor) sidecar
+that forwards incoming `.triggered` events to them.
+These distributor sidecars can also be used to send `.started` and `.finished` events
```

Comment:
I think in the meanwhile, two options for that exist, the distributor sidecar but there's also a control plane connector library.

Reply:
@thisthat Where could I find some info about this control plane connector?  And how to reword this?

Reply:
My understanding in terms of "maturity, when they were introduced and functionality":

1) distributor (the original mechanism)
2) [cp-connector (control plane connector)](https://github.com/keptn/go-utils/tree/master#creating-a-keptn-service-using-cp-connector) handles the logic of an integration connecting back to the control plane. Benefit is that it's handled in Go code and the distributor pod is no longer required. Drawback - you're coding it. Currently used by all "core" services.
3) [go-sdk](https://github.com/keptn/go-utils/tree/master#creating-a-keptn-service-using-the-go-sdk-experimental). [experimental] A wrapper around the cp-connector which adds features.

For me, this is very deep technical knowledge and shouldn't really live in the main docs?

Reply:
I think this sort of information very much belongs in the architecture document for the product, at some level.  I just incorporated Adam's explanation, which could be expanded but it clarifies things a lot.  We talk about cp-connector and go-sdk on Slack and I hear lots of comments about the distributor eventually going away plus they have been mentioned in the Release Notes.  It seems misleading to only mention the distributor. 

See what you think of my rewrite.  The Architecture section needs some serious attention and I didn't really mean to do it in this PR.  This does incrementally improve the information that was there.